docs/solidity-style.md

```diff
@@ -236,9 +240,9 @@ At the start of the Interactions block in a CEI-pattern function, set them off v
 
 When a function is an interaction made reentrancy-safe by the CEI pattern, follow its `@custom:interaction` mark with `CEI`, or with `RCEI` (R is for "Refresh") if it starts by calling `AssetRegistry.refresh()`.
 
-#### ReentrancyGuard
+#### ReentrancyGuard (Global Lock)
 
-Where using the CEI pattern is impractical, every function on that contract that is `external`, and can write to the relevant state elements, should use `reentrancyGuard`. That is, the contract should inherit from either `ReentrancyGuard` (or `ReentrancyGuardUpgradable` as needed), and every external function that can either modify contract state, or read it when it's inconsistent, should be marked with the `nonReentrant` modifier.
+Where using the CEI pattern is impractical, every function that is `external`, and can write to the relevant state elements, should use a global lock implemented via `GlobalReentrancyGuard` on `Main`. Every external function on individual components that can either modify contract state, or read it when it's inconsistent, should be marked with the `globalNonReentrant` modifier.
```

Comment:
This is not quite true, for example, `Furnace.melt()` isn't globalNonReentrant even though it modifies contract state. The prior text was wrong, though, you didn't add anything to break it. 

Instead let's say:
```
...via `GlobalReentrancyGuard` on `Main`. Every `@custom:interaction` should be market with the `globalNonReentrant` modifier.
```